docs/standard/datetime/timeprovider-overview.md

```diff
@@ -0,0 +1,87 @@
+---
+title: What is the TimeProvider class
+description: Learn about the TimeProvider class in .NET and .NET Framework. TimeProvider provides an abstraction over time.
+ms.date: 12/03/2024
+ms.topic: overview
+dev_langs:
+  - "csharp"
+  - "vb"
+helpviewer_keywords:
+  - "TimeProvider class"
+  - "date and time classes [.NET]"
+#customer intent: As a developer, I want to understand what TimeProvider is so that I can use it.
+---
+
+# What is TimeProvider?
```

Comment:
Could you please add a small section telling in .NET 8.0, we have added the following APIs that work with TimeProvider:

- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.-ctor?view=net-9.0#system-threading-cancellationtokensource-ctor(system-timespan-system-timeprovider](https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.-ctor?view=net-9.0#system-threading-cancellationtokensource-ctor(system-timespan-system-timeprovider))
- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-9.0#system-threading-tasks-task-delay(system-timespan-system-timeprovider](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-9.0#system-threading-tasks-task-delay(system-timespan-system-timeprovider))
- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-9.0#system-threading-tasks-task-delay(system-timespan-system-timeprovider-system-threading-cancellationtoken](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.delay?view=net-9.0#system-threading-tasks-task-delay(system-timespan-system-timeprovider-system-threading-cancellationtoken))
- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0#system-threading-tasks-task-waitasync(system-timespan-system-timeprovider](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0#system-threading-tasks-task-waitasync(system-timespan-system-timeprovider))
- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0#system-threading-tasks-task-waitasync(system-timespan-system-timeprovider-system-threading-cancellationtoken](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.waitasync?view=net-8.0#system-threading-tasks-task-waitasync(system-timespan-system-timeprovider-system-threading-cancellationtoken))

And on .NET Framework, can achieve the same functionality using the TimeProvider extension methods:

- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.timeprovidertaskextensions.createcancellationtokensource?view=net-8.0-pp](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.timeprovidertaskextensions.createcancellationtokensource?view=net-8.0-pp)
- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.timeprovidertaskextensions.delay?view=net-8.0-pp](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.timeprovidertaskextensions.delay?view=net-8.0-pp)
- [https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.timeprovidertaskextensions.waitasync?view=net-8.0-pp](https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.timeprovidertaskextensions.waitasync?view=net-8.0-pp)

These APIs will make it easy to the create mock up test using TimeProvider and consume such APIs to test any depend on such APIs.